File: frontend/src/types/order.ts
export type OrderItem = {
  productId: number;
  productName: string;
  price: number;
  quantity: number;
  lineTotal: number;
};

export type OrderResponse = {
  orderId: number;
  orderDate: string;
  status: string;
  total: number;
  shippingAddress: string;
  confirmationNumber: string;
  items: OrderItem[];
};

File: api/Models/Order.cs
namespace api.Models
{
    public class Order
    {
        public int Id { get; set; }
        public string UserId { get; set; } = "";
        public DateTime OrderDate { get; set; }
        public string Status { get; set; } = "";
        public decimal Total { get; set; }
        public string ShippingAddress { get; set; } = "";
        public string ConfirmationNumber { get; set; } = "";

        public List<OrderItem> Items { get; set; } = new();
    }
}